### ISSUE TITLE
Pre-release

### ISSUE THREAD

**Issue by brandonguigo** [chaaz/versio#28]
Hello @chaaz,

I would like to know how to use versio to do pre-releases for each PR. We have a monorepo containing all our apis and would like to build a pre-release for the project that the commits targets, which include a version bump like the one done in the release part but also a `rc.xxx` tag where xx is the rc number (increments for each PR)...

Is it possible with Versio ? 

Thanks

**Comment by robko23** [comment]
Hi. This is currently not possible. We don't have full implementation of semver, it's built for `MAJOR.MINOR.PATCH`.  (it could parse the RC string, but i don't think it would increment it correctly.
Howeve, I don't think I understand how the proposed changes should work. Every merged PR should increase RC number? Could you describe it a little more?

**Comment by chaaz** [comment]
Yes, I'm curious how that works: does each project have its own RC number that increments when a pre-release is done? Do only projects that have bumped their version have a pre-release process done? Does each PR invoke the pre-release process, or is it manually triggered?

I think Versio can currently help, but it probably can't do everything you're describing right now. I'm curious about your use-case, though.

**Comment by brandonguigo** [comment]
Hello @chaaz @robko23, sorry, been very busy at work and forgot to reply...

The workflow is the following: 
- we start work on a branch, create a PR
- This PR triggers a GH action workflow that test, and build a pre-release

To build this pre-release, we actually use the commit hash.  The ideal approach would be to use the `versio release` command with hooks to release the rc.

I think the process of bumping could be :
- if you are in stable mode (ie `1.0.3`) and you do a fix, you’ll end at `1.0.4-rc1` with for example a `—rc` flag. Versio would have to bump normally but add a `-rc1` suffix
- If you are in RC channel (ie `1.0.4-rc1`), in a PR, you’ll end up in `1.0.4-rc2`

I can answer questions whenever you need (and will be more available to work with you !)
